app/models/user.rb

```diff
@@ -466,9 +466,9 @@ def banished?
   end
 
   def subscribe_to_mailchimp_newsletter
-    return unless registered
-    return unless email.present? && email.include?("@")
-    return if saved_changes["unconfirmed_email"] && saved_changes["confirmation_sent_at"]
+    return unless registered && email.present?
+    return if SiteConfig.mailchimp_api_key.blank? && SiteConfig.mailchimp_newsletter_id.blank?
+    return if saved_changes.key?(:unconfirmed_email)
     return unless saved_changes.key?(:email) || saved_changes.key?(:email_newsletter)
```

Comment:
I refactored this to 4 guard clauses but it's still a lot IMO 🤔

Reply:
Won't the removal of `&& saved_changes["confirmation_sent_at"]` change the logic somehow?

Reply:
Ah, you're right! Added it back in. I thought about it and it seemed okay at first but I should've went through and checked it properly.